src/format/opamFormat.ml

```diff
@@ -245,6 +245,20 @@ module Pp = struct
     name_constr: string -> string;
   }
 
+  let map_right f g t = {
+    parse = (fun ~pos x -> f (t.parse ~pos x));
+    print = (fun x -> t.print (g x));
+    name = t.name;
+    name_constr = t.name_constr;
+  }
+
+  let map_left f g t = {
+    parse = (fun ~pos x -> t.parse ~pos (g x));
+    print = (fun x -> f (t.print x));
+    name = t.name;
+    name_constr = t.name_constr;
+  }
+
```

Comment:
See above, I think you can define a pp with the two functions and use the normal pipe operator `-|` for the same result.